src/components/authorize/jwt-service.ts

```diff
@@ -29,13 +30,55 @@ export class JwtService implements JwtServiceInterface {
     }
   }
 
-  getPayload(jwt: string): AuthorizeRequestPayload {
+  async getPayloadWithSigCheck(jwt: string): Promise<AuthorizeRequestPayload> {
     const jwtElements = jwt.split(".");
     if (jwtElements.length !== 3) {
       throw new JwtPayloadParseError("JWT was not three elements");
     }
+
+    if ((await this.signatureCheck(jwt)) === false) {
+      throw new JwtSignatureVerificationError("Jwt Signature is not valid");
+    }
     const payload = jwtElements[1];
     const buffer = Buffer.from(payload, "base64url");
+
     return JSON.parse(buffer.toString());
   }
+
+  validateClaims(claims: AuthorizeRequestPayload): AuthorizeRequestPayload {
+    const expectedkeys = [
+      "iss",
+      "aud",
+      "exp",
+      "iat",
+      "nbf",
+      "jti",
+      "client_name",
+      "cookie_consent_shared",
+      "consent_required",
+      "is_one_login_service",
+      "service_type",
+      "govuk_signin_journey_id",
+      "confidence",
+      "state",
+      "client_id",
+      "scope",
+      "redirect_uri",
+    ];
+    expectedkeys.forEach((claim) => {
+      if (!Object.prototype.hasOwnProperty.call(claims, claim)) {
+        throw new ClaimsError(`${claim} claim missing`);
+      }
+    });
+
+    if (claims.exp <= Math.floor(new Date().getTime() / 1000)) {
```

Comment:
In the AuthRequest it explicitly sets it to UTC. Do we need to explicit here and use UTC?

Reply:
The date is passed across as unix time, and is validated using unix time. We shouldn't need worry about timezones?

Reply:
If that is the case, then great